client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -122,6 +122,15 @@ public synchronized Schema getBySubjectAndID(String subject, int id)
   }
 
   @Override
+  public SchemaMetadata getSchemaMetadata(String subject, int version) throws IOException, RestClientException {
+    io.confluent.kafka.schemaregistry.client.rest.entities.Schema response
+    = restService.getVersion(subject, version);
+   int id = response.getId();
+   String schema = response.getSchema();
+   return new SchemaMetadata(id, version, schema);
+  }
+
+  @Override
```

Comment:
Do you want to give the same treatment to this version as you did in `MockSchemaRegistryClient` where you don't repeat the code and instead reuse the `getSchemaMetadata(String)` method?